backend/inspirehep/matcher/parsers.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+from inspire_dojson.utils import strip_empty_values
+from inspire_schemas.api import ReferenceBuilder
+from parsel import Selector
+
+
+class GrobidParser:
+    def __init__(self, content):
+        self.builder = ReferenceBuilder()
+        self.root = Selector(text=content, type="xml")
+
+    def parse(self):
+        for report_number in self.report_numbers:
+            self.builder.add_report_number(report_number)
+
+        self.builder.add_uid(self.isbn)
+        self.builder.add_uid(self.arxiv_eprint)
+
+        for doi in self.dois:
+            self.builder.add_uid(doi)
+
+        self.builder.set_journal_issue(self.journal_issue)
+        self.builder.set_journal_volume(self.journal_volume)
+        self.builder.set_journal_title(self.journal_title)
+        self.builder.set_page_artid(page_start=self.page_start, page_end=self.page_end)
+        self.builder.set_year(self.year)
+        return strip_empty_values(self.builder.obj)
```

Comment:
is the `strip_empty_values` really needed?

Reply:
Yes, otherwise it returns `None` values in the keys, it should have been handled in the ``ReferenceBuilder`` I guess

Reply:
:cry: